tests/squirrel.test.ts

```diff
@@ -0,0 +1,88 @@
+import Squirrel from "../src/animals/Squirrel";
+
+test("should log an error if  squirrel age is negative", () => {
+    expect(() => {
+        const lion = new Squirrel("Kiki", -5, true, "Cedar", 5, 8);
+    }).toThrow("Age cannot be negative");
+});
+
+test("should log an error if name is an empty string", () => {
+    expect(() => {
+        const lion = new Squirrel("", 5, true, "Cedar", 5, 8);
+    }).toThrow("Name cannot be empty");
+});
+
+test("should log an error if treeAge is negative", () => {
+    expect(() => {
+        const squirrel = new Squirrel("Kiki", 5, true, "Cedar", -5, 56);
+    }).toThrow("Tree age must be greater than zero");
+});
+
+test("should log an error if holeSize is negative", () => {
+    expect(() => {
+        const squirrel = new Squirrel("Kiki", 5, true, "Cedar", 5, -56);
+    }).toThrow("Hole size must be greater than zero");
+});
+
+describe("Squirrel Class Tests", () => {
+    test("should create a Squirrel instance with valid properties", () => {
+        const squirrel = new Squirrel("Kiki", 5, true, "Cedar", 5, 8);
+
+        expect(squirrel.getName()).toBe("Kiki");
+        expect(squirrel.getAge()).toBe(5);
+        expect(squirrel.getIsMale()).toBe(true);
+        expect(squirrel.getHome()).toBe("Cedar tree"); // ------tree ili Cedar da e home
+    });
+
+    test("should call walk method correctly", () => {
+        const logLock = jest.spyOn(console, "log").mockImplementation(() => { });
+        const squirrel = new Squirrel("Kiki", 5, true, "Cedar", 5, 8);
+        const walkSpy = jest.spyOn(squirrel, "walk");
+        squirrel.walk();
```

Comment:
what is the logic behing callling a walk method and then spying if the walk method was called? 
It might be a good idea in a complex method that has a couple of "if"s to check which codepath was taken based on the input parameters and thus check which internal method was called.